library/packages/src/modules/PackageCallbacks.rb

```diff
@@ -2487,6 +2487,110 @@ def ProcessDone
       nil
     end
 
+    # handle the file conflict detection start callback
+    def FileConflictStart
+      log.info "Starting file conflict check..."
+      # TRANSLATORS: progress bar label
+      label = _("Checking file conflicts...")
+
+      if Mode.commandline
+        CommandLine.PrintVerbose(label)
+      elsif UI.WidgetExists(:progressCurrentPackage)
+        # package slideshow with progress already present
+        UI.ChangeWidget(Id(:progressCurrentPackage), :Value, 0)
+        UI.ChangeWidget(Id(:progressCurrentPackage), :Label, label)
+      else
+        # TRANSLATORS: help text for the file conflict detection progress
+        help = _("<p>Detecting the file conflicts is in progress.</p>")
+        Progress.Simple(label, "", 100, help)
+      end
+    end
+
+    # Handle the file conflict detection progress callback.
+    # @param progress [Fixnum] progress in percents
+    # @return [Boolean] true = continue, false = abort
+    def FileConflictProgress(progress)
+      log.debug "File conflict progress: #{progress}%"
+
+      sleep(1)
+
+      if Mode.commandline
+        CommandLine.PrintVerboseNoCR("#{CLEAR_PROGRESS_TEXT}#{progress}%")
+      elsif UI.WidgetExists(:progressCurrentPackage)
+        UI.ChangeWidget(Id(:progressCurrentPackage), :Value, progress)
+      else
+        UI.ChangeWidget(Id(:pb), :Value, progress)
+      end
+
+      ui = UI.PollInput
+      ui != :abort && ui != :cancel
+    end
+
+    # Handle the file conflict detection result callback.
+    # Ask to user whether to continue. In the AutoYaST mode an error is reported
+    # but the installation will continue ignoring the confliucts.
+    # @param excluded_packages [Array<String>] packages ignored in the check
+    #   (e.g. not available for check in the download-as-needed mode)
+    # @param conflicts [Array<String>] list of translated descriptions of
+    #   the detected file conflicts
+    # @return [Boolean] true = continue, false = abort
+    def FileConflictReport(excluded_packages, conflicts)
+      log.info "Excluded #{excluded_packages.size} packages in file conflict check"
+      log.debug "Excluded packages: #{excluded_packages.inspect}"
+      log.info "Found #{conflicts.size} conflicts: #{conflicts.join("\n\n")}"
+
+      # just continue installing packages if there is no conflict
+      return true if conflicts.empty?
+
+      # don't ask in autoyast or command line mode, just report/log the issues and continue
+      if Mode.autoinst || Mode.autoupgrade || Mode.commandline
```

Comment:
you can and should use `Mode.auto`  https://github.com/yast/yast-yast2/blob/630cf98148facf439a4439c189d610ba031d1537/library/general/src/modules/Mode.rb#L288